src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { client } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  postId?: number;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId = 0,
+  setComments,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [nameError, setNameError] = useState<boolean>(false);
+  const [emailError, setEmailError] = useState<boolean>(false);
+  const [commentTextError, setCommentTextError] = useState<boolean>(false);
```

Comment:
All errors could be combined into one state = { nameError, emailError, commentTextError }
Also this comment convenient for `user` data (email, name .....)